### DISCUSSION TITLE
Provider config option to control behavior of conditional re-registered field's value?

### DISCUSSION THREAD

**Discussion by miusuncle** [react-hook-form/react-hook-form#5670] [Q&A]
**Is your feature request related to a problem? Please describe.**
Since v7.8.4, conditional re-registered field's value will be erased (see https://codesandbox.io/s/wild-waterfall-r4lls?file=/src/App.js) no matter if default value had been provided. This diversity behavior stops me from upgrading to a new version.

**Describe the solution you'd like**
So I wondered if any option could be provided to solve this issue. Something like:

- Erase re-registered field's value like v7.8.4 did
- Recover re-registered field's value to default value like v7.8.3 did before
- Recover re-registered field's value to its latest value (if possible?)

**Comment by bluebill1049** [comment]
can you provide some more detail? which issue related to this?

**Reply by miusuncle** [reply]
Imagine we have a lot of fields rendered by condition (`shouldUnregistered` set to true of course), it is convenient to provide default values for these fields to save a lot of typing. It works pretty good before v7.8.4. After upgrading to v7.8.4 or lastest version(v7.8.8 at this point), the behavior changed (You know)，default values just don't works. And the convenient no longer exists（苦涩）.

**Reply by bluebill1049** [reply]
U mean toggle the input and retain its defaultValue for each input?

**Reply by miusuncle** [reply]
Yes, that was v7.8.3 or before did (see https://codesandbox.io/s/sad-chandrasekhar-n15ot). From practical perspective, it works best if recover to latest value when toggled.

**Reply by bluebill1049** [reply]
shouldn't you be using shouldUnregister: false then?

**Reply by miusuncle** [reply]
I tend to use `shouldUnregistered: true`, because when submitting, unrendered fields should not be collected in natural.
